azure-pipelines.yml

```diff
@@ -162,112 +171,113 @@ steps:
+      Microsoft.CookiecutterTools.dll
+      Microsoft.IronPythonTools.Resolver.pdb
+      Microsoft.IronPythonTools.Resolver.dll
+      Microsoft.Python.Analysis.Engine.dll
+      Microsoft.Python.Analysis.Engine.pdb
+      Microsoft.Python.LanguageServer.Core.dll
+      Microsoft.Python.LanguageServer.Core.pdb
+      PyDebugAttach*.pdb
+      PyDebugAttach*.dll
+      VsPyProf*.pdb
+      VsPyProf*.dll
+    TargetFolder: '$(Build.ArtifactStagingDirectory)/symbols_and_binaries'
+
+# Index sources and publish symbols if not a pull request
+- task: PublishSymbols@2
+  displayName: 'Publish symbols'
+  condition: notin(variables['Build.Reason'], 'PullRequest')
+  inputs:
+    symbolsFolder: '$(Build.ArtifactStagingDirectory)/symbols_and_binaries'
+    searchPattern: '*.pdb'
+    symbolServerType: TeamServices
+
+# MicroBuild cleanup
+- task: MicroBuildCleanup@1
+  displayName: 'Execute cleanup tasks'
+  condition: succeededOrFailed()
+
+# Compress staging output
+- task: ArchiveFiles@2
+  displayName: 'Compress staging dir'
+  condition: notin(variables['Build.Reason'], 'PullRequest')
+  inputs:
+    rootFolderOrFile: '$(Build.ArtifactStagingDirectory)'
+    includeRootFolder: false
+    archiveType: 'zip'
+    archiveFile: '$(Build.ArtifactStagingDirectory)/PTVSBuildArchive.zip'
+    replaceExistingArchive: true
+
+# Compress just the test data
+- task: ArchiveFiles@2
+  displayName: 'Compress test data dir'
+  inputs:
+    rootFolderOrFile: '$(Build.ArtifactStagingDirectory)/src/Python/Tests/TestData'
+    includeRootFolder: false
+    archiveType: 'zip'
+    archiveFile: '$(Build.ArtifactStagingDirectory)/TestData.zip'
+    replaceExistingArchive: true
 
-  # Publish build artifact: staging output
-  - task: PublishBuildArtifacts@1
-    displayName: 'Upload staging zip to build artifacts'
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/PTVSBuildArchive.zip'
-      ArtifactName: 'PTVS Staging Output'
+# Publish build artifact: binaries
+- task: PublishBuildArtifacts@1
+  displayName: 'Upload binaries to build artifacts'
+  inputs:
+    PathtoPublish: '$(Build.BinariesDirectory)/raw/binaries'
+    ArtifactName: Binaries
 
-  # Publish test data archive (so we don't have to pull down everything to run tests)
-  - task: PublishBuildArtifacts@1
-    displayName: 'Upload test data zip to build artifacts'
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/TestData.zip'
-      ArtifactName: 'TestData'
+# Publish build artifact: staging output
+- task: PublishBuildArtifacts@1
+  condition: notin(variables['Build.Reason'], 'PullRequest')
+  displayName: 'Upload staging zip to build artifacts'
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)/PTVSBuildArchive.zip'
+    ArtifactName: 'PTVS Staging Output'
+
+# Publish test data archive (so we don't have to pull down everything to run tests)
+- task: PublishBuildArtifacts@1
+  displayName: 'Upload test data zip to build artifacts'
```

Comment:
This testData artifact is used by the build pipeline here:
https://devdiv.visualstudio.com/DevDiv/_taskgroup/a7b18468-aab5-4be7-b718-02f85b0828c4

Reply:
I should probably add that as a comment in here so people know how it's used.